.github/workflows/build-workflow.yml

```diff
@@ -2,67 +2,35 @@ name: build-workflow
 
 on:
   push:
-    branches: [ main, 'release-*' ]
+    branches: [ main ]
+    tags:
+      - "*"
+
   workflow_dispatch:
+    inputs:
+      disable_cache:
+        description: Disable rust caching
+        type: boolean
+        required: false
+        default: false
 
 env:
   CARGO_TERM_COLOR: always
   RUSTFLAGS: -D warnings
 
 jobs:
-  build-amd64:
-    name: Build and publish for amd64
-    runs-on: Ubuntu-20.04
-
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-
-      - name: Enable toolchain via github action
-        uses: dtolnay/rust-toolchain@1.64.0
-
-      - name: Enable cache
-        # https://github.com/marketplace/actions/rust-cache
-        uses: Swatinem/rust-cache@v2
-
-      - name: Build packages for amd64
-        run: bash -x ./ci/build_scripts/build.sh
-
-      - name: Upload artifacts as zip
-        # https://github.com/marketplace/actions/upload-a-build-artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: debian-packages-amd64
-          path: target/debian/*.deb
-
-      - name: Publish debian packages (for internal use)
-        env:
-          PUBLISH_URL: ${{ secrets.PUBLISH_URL }}
-          PUBLISH_TOKEN: ${{ secrets.PUBLISH_TOKEN }}
-          PUBLISH_REPO: ${{ secrets.PUBLISH_REPO }}
-          PUBLISH_DISTRIBUTION: ${{ secrets.PUBLISH_DISTRIBUTION }}
-        run: |
-          VERSION=$(./ci/build_scripts/build.sh --version | tail -1)
-          bash -x ./ci/build_scripts/publish_debian.sh \
-            --path target/debian/ \
-            --distribution "tedge-main" \
-            --component "main" \
-            --group "main/$VERSION/amd64"
-
-  build-arm-matrix:
+  build:
     name: Build and publish for ${{ matrix.job.target }}
     runs-on: Ubuntu-20.04
     strategy:
+      fail-fast: false
```

Comment:
Do you want to continue even if some build fails?

Reply:
yes (imho) as the builds are conducted in parallel, so abruptly cancelling the other parallel jobs when one fails just makes it more complicated to see which jobs have a problem.

Reply:
Plus we are only "building" it and not running tests, so there is more to gain by letting each job finish independently

Reply:
Ok, but while uploading does it miss one of the packages?

Reply:
Yes technically it could result in publishing one architecture and not the other...though that could have happened before (as there is no guarantee that another build will fail sooner than another target can publish it).

Though it maybe it might make sense to move the publishing to a new matrix job, which is dependent on all targets building successful (so the behaviour is a little more predictable)

Reply:
Agree, it's okay for now, We can update this in a follow-up PR.

Reply:
I just pushed changes to split the `build` and `publish` steps, and `publish` is dependent on the `build` job being successful.

Reply:
Nice, thanks for addressing this.